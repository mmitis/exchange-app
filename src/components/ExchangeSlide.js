//@flow
import React from "react";

type Props = {
    onLeftClick: () => void,
    onRightClick: () => void,
    children: React$Node,
    className: string
}
export const ExchangeSlide = ({
                                  onLeftClick = () => { },
                                  onRightClick = () => { },
                                  className = '',
                                  children
} : Props) => {
    return (
            <div className={`exchange-slide ${className}`}>
                    <button className="exchange-slide__button" onClick={onLeftClick}>
                        <i className="material-icons">keyboard_arrow_left</i>
                    </button>
                   
                    <div>
                        {children}
                    </div>
                    <button className="exchange-slide__button" onClick={onRightClick}>
                        <i className="material-icons">keyboard_arrow_right</i>
                    </button>
            </div>
    )
};

export default React.memo(ExchangeSlide);
