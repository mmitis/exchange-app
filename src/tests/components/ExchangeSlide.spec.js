import ExchangeSlide from './../../components/ExchangeSlide';
import { mount } from 'enzyme';
import React from "react";

describe('Component: ExchangeSlide', () => {

    it('test simple render', async () => {
        const props = {
            onLeftClick: () => {},
            onRightClick: () => {},
            className: 'sample-class'
        };
        const rotate = mount(<ExchangeSlide
            {...props}
        >
            <div className={'sample-div'}></div>

        </ExchangeSlide>);
        
        expect(rotate).toMatchSnapshot();
    });

});
