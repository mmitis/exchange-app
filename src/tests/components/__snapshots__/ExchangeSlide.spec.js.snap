// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ExchangeSlide test simple render 1`] = `ReactWrapper {}`;
